src/Contract/ClaimAuctionLotBidder.purs

```diff
@@ -0,0 +1,289 @@
+    mkAuctionToken tokenName = Value.singleton auctionCs tokenName one
+
+    -- AuctionEscrow -------------------------------------------------
+
+    auctionEscrowOref :: TransactionInput
+    auctionEscrowOref = fst auctionEscrowUtxo
+
+    auctionEscrowRedeemer :: Redeemer
+    auctionEscrowRedeemer = wrap $ toData BidderBuysRedeemer
+
+    auctionEscrowDatum :: Datum
+    auctionEscrowDatum = wrap $ toData AuctionConcluded
+
+    auctionEscrowValue :: Value
+    auctionEscrowValue = mkAuctionToken auctionEscrowTokenName
+      <> mkAuctionToken standingBidTokenName
+
+    -- StandingBid ---------------------------------------------------
+
+    standingBidOref :: TransactionInput
+    standingBidOref = fst standingBidUtxo
+
+    standingBidRedeemer :: Redeemer
+    standingBidRedeemer = wrap $ toData ConcludeAuctionRedeemer
+
+    --
+
+    totalAuctionFeesValue :: Value
+    totalAuctionFeesValue = Value.lovelaceValueOf $ totalAuctionFees auctionTerms
+
+    sellerPayoutValue :: Value
+    sellerPayoutValue = Value.lovelaceValueOf $ sellerPayout auctionTerms bidTerms
+
+    txValidRange :: POSIXTimeRange
+    txValidRange =
+      mkFiniteInterval nowTime
+        (auctionTermsRec.purchaseDeadline - wrap (BigInt.fromInt 1000))
+
+    constraints :: TxConstraints Void Void
+    constraints = mconcat
+      [ -- Spend auction escrow utxo:
+        Constraints.mustSpendScriptOutput auctionEscrowOref auctionEscrowRedeemer
+
+      , -- Spend standing bid utxo:
+        Constraints.mustSpendScriptOutput standingBidOref standingBidRedeemer
+
+      , -- Lock auction escrow and standing bid tokens at auction
+        -- escrow validator address:
+        Constraints.mustPayToScript validatorHashes.auctionEscrow auctionEscrowDatum
+          DatumInline
+          auctionEscrowValue
+
+      , -- Send auction lot to the buyer:
+        Constraints.mustPayToPubKey buyerPkh auctionTermsRec.auctionLot
+
+      , -- Send auction proceeds to the seller:
+        Constraints.mustPayToPubKey sellerPkh sellerPayoutValue
+
+      , -- Lock total auction fees at fee escrow validator address:
+        Constraints.mustPayToScript validatorHashes.feeEscrow unitDatum DatumInline
+          totalAuctionFeesValue
+
+      , -- This transaction must be signed by the buyer:
+        Constraints.mustBeSignedBy buyerPkh
+
+      , -- Set transaction validity interval to purchase period:
+        Constraints.mustValidateIn txValidRange
+      ]
+
+    lookups :: ScriptLookups Void
+    lookups = mconcat
+      [ Lookups.unspentOutputs $ Map.fromFoldable [ auctionEscrowUtxo, standingBidUtxo ]
+      , Lookups.validator (unwrap validators).auctionEscrow
+      , Lookups.validator (unwrap validators).standingBid
+      ]
+
+  lift $ submitTxReturningContractResult {} $ emptySubmitTxData
+    { lookups = lookups
+    , constraints = constraints
+    }
+
+queryStandingBidUtxo :: AuctionInfo -> Contract (Maybe (Utxo /\ StandingBidState))
```

Comment:
We already have `queryStandingBidUtxo` with the same signature, though I see its logic is different. But for the clearness' sake we might want to find another name (not sure, I will re-read it both functions more carefully).

Reply:
Very good catch! Extracted utxo queries into a separate module `HydraAuctionOffchain.Contract.QueryUtxo` and eliminated the mentioned duplication.